vllm/v1/engine/core_client.py

```diff
@@ -134,6 +137,20 @@ def __init__(
         executor_class: Type[Executor],
         log_stats: bool = False,
     ):
+        # The child processes will send SIGUSR1 when unrecoverable
+        # errors happen. We kill the process tree here so that the
+        # stack trace is very evident.
+        # TODO(rob): rather than killing the main process, we should
+        # figure out how to raise an AsyncEngineDeadError and
+        # handle at the API server level so we can return a better
+        # error code to the clients calling VLLM.
+        def sigusr1_handler(signum, frame):
+            logger.fatal(
+                "Got SIGUSR1 from worker processes, shutting "
+                "down. See stack trace above for root cause issue.")
```

Comment:
nit: maybe we want to do something like this, just because SIGUSR1 is a little esoteric
```suggestion
            logger.fatal(
                "Got fatal signal from worker processes, shutting "
                "down. See stack trace above for root cause issue.")
```